### PR TITLE
Add basic monitoring test in SLES4SAP

### PR DESCRIPTION
**Pull request by ldevulder**
SUSE has implemented some monitoring tools for HA, HANA and NW server/cluster.

This commit adds basic monitoring check.

- Related ticket: N/A
- Needles: N/A
- Verification runs: [netweaver_node01](http://1b210.qa.suse.de/tests/8814), [netweaver_node02](http://1b210.qa.suse.de/tests/8815), [hana_node01](http://1b210.qa.suse.de/tests/8829), [hana_node02](http://1b210.qa.suse.de/tests/8830)

**EDIT:** new verification runs: [netweaver_node01](http://1b210.qa.suse.de/tests/8895), [netweaver_node02](http://1b210.qa.suse.de/tests/8896), [hana_node01](http://1b210.qa.suse.de/tests/8901), [hana_node02](http://1b210.qa.suse.de/tests/8902)

### CODE REVIEW

File: tests/sles4sap/monitoring_services.pm
Comment (alvarocarvajald) [inline tests/sles4sap/monitoring_services.pm:153]: It could be useful to save the metrics to a file and upload it to the test.

File: tests/sles4sap/monitoring_services.pm
Comment (alvarocarvajald) [inline tests/sles4sap/monitoring_services.pm:152]: This is the first initialization of this variable, so I think a simple `=` would be better. No need to use `//=`

File: tests/sles4sap/monitoring_services.pm
Comment (alvarocarvajald) [inline tests/sles4sap/monitoring_services.pm:48]: I think this whole block from L48 to L93 could be a separate method/function within the module. It could improve readability.

File: tests/sles4sap/monitoring_services.pm
Comment (alvarocarvajald) [inline tests/sles4sap/monitoring_services.pm:96]: Same here from L96 to L141.

File: tests/sles4sap/monitoring_services.pm
Comment (alvarocarvajald) [inline tests/sles4sap/monitoring_services.pm:113]: s/create/created

File: tests/sles4sap/monitoring_services.pm
Comment (alvarocarvajald) [inline tests/sles4sap/monitoring_services.pm:41]: Not sure if it would be better here to use `||=` instead of `//=`. For example, a port value of 0 in the config file would be evaluated as true by `//=` and not change the value on `$ha_export_port`, and a port value of 0 further down the test would not yield valid results, but in a configuration file 0 could mean "use the default".

File: tests/sles4sap/monitoring_services.pm
Comment (alvarocarvajald) [inline tests/sles4sap/monitoring_services.pm:86]: Same comment here WRT `//=` vs `||=`

File: tests/sles4sap/monitoring_services.pm
Comment (alvarocarvajald) [inline tests/sles4sap/monitoring_services.pm:133]: Same comment here WRT `//=` vs `||=`

**Review by alvarocarvajald — APPROVED** [review]
LGTM in general. I'm leaving a few suggestions on minor things.

File: tests/sles4sap/monitoring_services.pm
Comment (ldevulder) [reply tests/sles4sap/monitoring_services.pm:152]: It was not the case in the first version, I forgot to remove this ;-)

File: tests/sles4sap/monitoring_services.pm
Comment (ldevulder) [reply tests/sles4sap/monitoring_services.pm:48]: Yes, I was thinking of this also. Will do it.

File: tests/sles4sap/monitoring_services.pm
Comment (ldevulder) [reply tests/sles4sap/monitoring_services.pm:41]: Thanks, in that case yes `||=` could be better.

File: tests/sles4sap/monitoring_services.pm
Comment (ldevulder) [reply tests/sles4sap/monitoring_services.pm:154]: Yes, I will add this.

File: tests/sles4sap/monitoring_services.pm
Comment (juadk) [inline tests/sles4sap/monitoring_services.pm:83]: Good idea, we should rework all of our cluster commands like that if it works on 12-SP2.

File: tests/sles4sap/monitoring_services.pm
Comment (juadk) [reply tests/sles4sap/monitoring_services.pm:49]: Yes, good point!

**Review by juadk — APPROVED** [review]
LGTM, I will review it again once the WIP flag removed.

File: tests/sles4sap/monitoring_services.pm
Comment (ldevulder) [reply tests/sles4sap/monitoring_services.pm]: Done.

File: tests/sles4sap/monitoring_services.pm
Comment (ldevulder) [reply tests/sles4sap/monitoring_services.pm]: Done.

File: tests/sles4sap/monitoring_services.pm
Comment (ldevulder) [reply tests/sles4sap/monitoring_services.pm]: Done.

File: tests/sles4sap/hana_install.pm
Comment (alvarocarvajald) [inline tests/sles4sap/hana_install.pm:193]: I think it would be better to move this to `lib/sles4sap::copy_media()` as it's there where the current directory is being changed. I don't believe `hdblcm` uses anything from a relative path.

**Review by alvarocarvajald — APPROVED** [review]
LGTM. Only a small comment.

File: tests/sles4sap/hana_install.pm
Comment (ldevulder) [reply tests/sles4sap/hana_install.pm:193]: Could be yes, I will check and test.

Comment (ldevulder) [reply tests/sles4sap/hana_install.pm:193]: It seems that NW installer doesn't like to be executed if we run it from another directory: http://1b210.qa.suse.de/tests/8868#step/netweaver_install/45

Comment (ldevulder) [reply tests/sles4sap/hana_install.pm]: Done!

Comment (ldevulder) [reply tests/sles4sap/hana_install.pm]: The latest change I made needs some modification in the jobs group YAML file, because I included the ARCH in the mount command.

I will make a MR for the BV and the maintenance tests before merging this PR.

**Review by alvarocarvajald — APPROVED** [review]
LGTM

File: lib/sles4sap.pm
Comment (alvarocarvajald) [inline lib/sles4sap.pm:354]: I went ahead and approved the PR, but I wanted to comment on this:

I think with this change we're losing some flexibility that was present in the old code, and I'm not sure the advantages of the change outweigh this. Basically, previously we could pass in **HANA** or **NW** a path such as `/path/to/installation/master` as well as `/path/to/installation/master/arch`. With both of those, code would still work, as the `type_string "cd " . get_var('ARCH') . "\n";` command would fail silently if there is not **ARCH** subdir in the share, but allows the test to continue.

With the new code, if we pass a share that has no **ARCH** subdir, we'll get a failure in the test, even if such a share points to a valid HANA or NW installation master.

Advantages I see in the new code are 2:
1. We remove a possible `cd` error in the test results, which again, this was mostly silent and didn't impact the outcome of the test.
2. We remove an interaction with SUT while copying.

But, we lose the flexibility of passing the test a share that it's not formatted with an **ARCH** subdir even if it's a valid share (for example, the shares in kvmmaster). I'm not sure if this is a good idea or not, but will not oppose the change.

Comment (juadk) [reply lib/sles4sap.pm:354]: I agree with you, there are pros and cons with both solutions. As Loic did the test like this, my point of view is to merge it as is. Loic! It's your choice now :smile:

**Review by juadk — APPROVED** [review]
LGTM

File: lib/sles4sap.pm
Comment (ldevulder) [reply lib/sles4sap.pm]: Ok, so I think it's better to keep the old behavior in that case. I just changed a little the logic to avoid unneeded `cd` command.
Let me know if it's OK for you.

Comment (alvarocarvajald) [reply lib/sles4sap.pm]: I like it. It preserves the logic, avoids doing a `cd` to determine the **ARCH** subdir exists so there's no silent error in the serial terminal and keeps the same number of interactions with SUT, but replacing a `type_string` with a `script_run` which is better for error handling.

Only change I'd propose would be with the `get_var('ARCH')` in L352: either to switch it to `get_required_var('ARCH')` or to give it a default with `get_var('ARCH', '')`.

Thanks for this change!

Comment (ldevulder) [reply lib/sles4sap.pm]: Last modification done! I will now restart the verification runs.

**Pull request merged**
